tests/auxtel/detector_characterization/test_std_flat_dataset.py

```diff
@@ -0,0 +1,81 @@
+import sys
+import unittest
+import asyncio
+import numpy as np
+import logging
+
+from lsst.ts import salobj
+
+from lsst.ts.scriptqueue import ScriptState
+from lsst.ts.standardscripts.auxtel.detector_characterization import ATGetStdFlatDataset
+
+import SALPY_ATCamera
+import SALPY_ATSpectrograph
+
+np.random.seed(47)
+
+index_gen = salobj.index_generator()
+
+logger = logging.getLogger()
+logger.level = logging.DEBUG
+
+
+class Harness:
+    def __init__(self):
+        self.index = next(index_gen)
+
+        self.test_index = next(index_gen)
+
+        salobj.test_utils.set_random_lsst_dds_domain()
+
+        self.script = ATGetStdFlatDataset(index=self.index, descr='Test ATGetStdFlatDataset')
+
+        # Adds controller to Test
+        self.at_cam = salobj.Controller(SALPY_ATCamera)
+        self.at_spec = salobj.Controller(SALPY_ATSpectrograph)
+
+        self.nbias = 0
+        self.ndark = 0
+        self.nflat = 0
+
+    async def cmd_take_images_callback(self, id_data):
+        if "bias" in id_data.data.imageSequenceName:
+            self.nbias += 1
+        elif "dark" in id_data.data.imageSequenceName:
+            self.ndark += 1
+        elif "flat" in id_data.data.imageSequenceName:
+            self.nflat += 1
```

Comment:
Really picky but...consider using the same variable names here as in the script. I like `nflat` just fine but it's a bit confusing to have `n_flat` in one place and `nflat`. Either is fine.